Lib/gftools/utils.py

```diff
@@ -278,6 +278,32 @@ def _html_custom_formatter(string):
     string = string.replace(". ", f".\n{whitespace}")
     string = string.replace("! ", f"!\n{whitespace}")
     string = string.replace("? ", f"?\n{whitespace}")
+    # Read into list
+    strings = string.split("\n")
+    # Cycle through list to find abbreviations
+    for i, s in enumerate(strings):
+        this_line = strings[i]
+        next_line = strings[i + 1] if i + 1 < len(strings) else ""
+        if next_line:
+            if (
+                re.search(r"i\.?e\.$", this_line)  # ie.
+                or re.search(r"e\.?g\.$", this_line)  # eg.
+                or (
+                    re.search(r"etc[\.|\?|!]$", this_line)
+                    and next_line[1] == next_line[1].lower()
+                )  # etc.
+                or (
+                    re.search(r"\W\w{1,2}[\.|\?|!]$", this_line)
+                    and this_line[-2] == this_line[-2].upper()
+                    and next_line[1] == next_line[1].lower()
+                )  # H.R. Giger
+            ):
+                strings[i] = strings[i] + strings[i + 1]
+                strings[i + 1] = ""
+    # Join back together
+    string = "\n".join(strings)
```

Comment:
We could use a range to make this part cleaner

```suggestion
    for i in range(1, len(strings):
        this_line = strings[i-1]
        next_line = strings[i]
        if next_line = "":
            continue
        if (
            re.search(r"i\.?e\.$", this_line)  # ie.
            or re.search(r"e\.?g\.$", this_line)  # eg.
            or (
                re.search(r"etc[\.|\?|!]$", this_line)
                and next_line[1] == next_line[1].lower()
            )  # etc.
            or (
                re.search(r"\W\w{1,2}[\.|\?|!]$", this_line)
                and this_line[-2] == this_line[-2].upper()
                and next_line[1] == next_line[1].lower()
            )  # H.R. Giger
        ):
            strings[i-1] = strings[i-1] + strings[i]
            strings[i] = ""
    # Join back together
    string = "\n".join(strings)
```

Reply:
As you prefer Marc. Feel free to rewrite it, as the test will ensure the correct formatting.

Reply:
Let me test this first.